test/run.d

```diff
@@ -186,19 +186,28 @@ Options:
         }
 
         bool failed;
+        string[] failedTargets;
         ensureToolsExists(env, EnumMembers!TestTools);
         foreach (target; parallel(targets, 1))
         {
             log("run: %-(%s %)", target.args);
             int status = spawnProcess(target.args, env, Config.none, scriptDir).wait;
             if (status != 0)
             {
-                writeln(">>> TARGET FAILED: ", target.filename);
+                const name = target.normalizedTestName;
+                writeln(">>> TARGET FAILED: ", name);
                 failed = true;
+                failedTargets ~= name;
             }
         }
         if (failed)
+        {
+            // print overview of failed targets (for CIs)
+            import std.ascii : newline;
+            writeln("FAILED targets:");
+            failedTargets.map!(l => chain("- ", l)).joiner(newline).writeln;
```

Comment:
Other processes should already be joined here, so you could probably get away with:

```suggestion
            writefln(`%-(- %s%)`, failedTargets);
```

Reply:
Fair enough. I used `.each!writeln` as I wanted to have a lineper test and that's IMHO the most readable.